scripts/yggdrasil-iptunnel/yggdrasil-setup

```diff
@@ -4,36 +4,137 @@
+        if [[ "${PUBLIC_KEY}" =~ ^[0-z]{64} ]]; then
+            yggdrasilctl addRoute subnet="${IP_ADDR}/32" box_pub_key="${PUBLIC_KEY}"
+            if [[ ! -z "${IP6_ADDR}" ]]; then
+                yggdrasilctl addRoute subnet="${IP6_ADDR}/128" box_pub_key="${PUBLIC_KEY}"
+                if [[ ! -z "${IP6_SUBNET}" ]]; then
+                    yggdrasilctl addRoute subnet="${IP6_SUBNET}" box_pub_key="${PUBLIC_KEY}"
+                    ip -6 route add "${IP6_SUBNET}" dev "${YGGINT}" || true
+                fi
+                isIPv6=true
+            fi
+        fi
+    done < /etc/yggdrasil.iptunnel.server
+
+    yggdrasilctl addSourceSubnet subnet="0.0.0.0/0"
+    ip route add "${SUBNET4}" dev "${YGGINT}" || true
+
+    if [[ "${isIPv6}" == true ]]; then
+        yggdrasilctl addSourceSubnet subnet="::/0"
+        # Add route for ygg0 interface
+        ip -6 route add "${SUBNET6}" dev "${YGGINT}" || true
+    fi
+
+    # If no NAT (masquarading) being done, start doing it
+    if [ -z "$(iptables -L POSTROUTING -v -n -t nat  | grep MASQUERADE)" ]; then
+        iptables -t nat -A POSTROUTING -o "${OUTINT}" -j MASQUERADE
+    fi
     # If no NAT (masquarading) being done, start doing it
-    if [ -z "$(sudo iptables -L POSTROUTING -v -n -t nat  | grep MASQUERADE)" ]; then
-        sudo iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+    if [ -z "$(ip6tables -L POSTROUTING -v -n -t nat  | grep MASQUERADE)" ]; then
+        if [[ ! "${IPv6NAT}" == false ]]; then
+            iptables -t nat -A POSTROUTING -o "${OUTINT}" -j MASQUERADE
+        fi
     fi
 # Configure as yggdrasil iptunnel client if client file is present (this is simply a newline-separated list
 # of cjdns public keys in /etc/yggdrasil.iptunnel.client, each key indicating an iptunnel exit server)
 elif [ -e /etc/yggdrasil.iptunnel.client ]; then
     # Add each server to yggdrasil iptunnel connect-to's
-    while read -r PUBLIC_KEY ASSIGNED_IP; do
+    while read -r PUBLIC_KEY IP_ADDR IP6_ADDR IP6_SUBNET; do
         if [[ "${PUBLIC_KEY}" =~ ^[0-z]{64} ]]; then
-            sudo ip addr add dev ygg0 ${ASSIGNED_IP}/32
+            if [[ ! -z "${IP6_ADDR}" ]]; then
+                ip addr add dev "${YGGINT}" "${IP6_ADDR}/128"
+                yggdrasilctl addRoute subnet="::0/0" box_pub_key="${PUBLIC_KEY}"
+                yggdrasilctl addSourceSubnet subnet="${IP6_ADDR}/128"
+                if [[ ! -z "${IP6_SUBNET}" ]]; then
+                        yggdrasilctl addSourceSubnet subnet="${IP6_SUBNET}"
+                        ip -6 route add ${IP6_SUBNET} dev "${IPv6SUBNETINT}"
+	                ip addr add dev "${IPv6SUBNETINT}" "$(echo  ${IP6_SUBNET} | sed "s/\/64/1/")"
+                fi
+            fi
+            ip addr add dev "${YGGINT}" "${IP_ADDR}/32"
+            yggdrasilctl addRoute subnet="0.0.0.0/0" box_pub_key="${PUBLIC_KEY}"
+            yggdrasilctl addSourceSubnet subnet="${IP_ADDR}/32"
```

Comment:
Won't these three commands try an add an IPv4 subnet, even if one doesn't exist? There should be another if, right?

Reply:
Also how do you know it's `/32`?

Reply:
IPv4 address ALWAYS needs to exist,  if it does not the tunnel will fall apart. Although TECHNICALLY you dont need ipv4 to work since this is meant for the whole internet, its kinda pointless without ipv4.

IPv6 is optional, IPv4 is not

`$IP_ADDR` is taken from the `yggdrasil.iptunnel.client` file

`/32` denotes only ONE ip address. This side of the tunnel (client side) has a network of only 1 ip address the server side is has all 256 with a `/24`

Reply:
It's a CIDR, ah okay.

I wonder if it makes sense to still support IPv6 only Internet, but I guess that is not actually used anywhere.